### PR TITLE
[sources] cache relative sources in redux store

### PR DESCRIPTION
**Pull request by jasonLaster**
Fixes Issue: #6774 

### Summary of Changes

This switches getRelativeSources from a memoized selector to a redux field. This is similar to how we now cached sources by url as well. The trade-off is memory vs computation. Ideally all the computed data would be cached in a selector, but that is difficult to do and maintain the same performance properties.

- [new version](https://perfht.ml/2vL57e7)
- [old version](https://perfht.ml/2vKElm9)

### old
image

### new
image

### CODE REVIEW

File: src/test/tests-setup.js [hunk 1]
```diff
@@ -80,6 +80,7 @@ beforeEach(() => {
   clearSymbols();
   clearHistory();
   clearDocuments();
+  prefs.projectDirectoryRoot = "";
```
Comment (jasonLaster) [inline src/test/tests-setup.js]: the fact that we weren't clearing prefs prior is pretty scary...

File: src/reducers/sources.js [hunk 2]
```diff
@@ -184,12 +191,49 @@ function updateSource(state: SourcesState, source: Object) {
...
+
+  const relativeSource = {
+    ...source,
+    relativeUrl: getRelativeUrl(source, root)
+  };
+
+  return {
+    ...state,
+    relativeSources: {
+      ...state.relativeSources,
```
Comment (AnshulMalik) [inline src/reducers/sources.js]: We call `updateRelativeSource` for every source. And here it seems it's doing a lot of work. Your opinion?

Comment (jasonLaster) [reply src/reducers/sources.js]: oh wow - good point. I didn't think of that!

Comment (jasonLaster) [reply src/reducers/sources.js]: we should definitely have a mutable helper here

Comment (jasonLaster) [reply src/reducers/sources.js]: just updated. what do you think?

Comment (AnshulMalik) [reply src/reducers/sources.js]: Looking good :)

**Pull request merged**